### PR TITLE
Fix the errors reported by NullAway

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

Before adopting NullAway(#1680) or other static analysis tools.
Manually run it and clean up errors.

Modifications:

- Fix all true positives reported by NullAway

Result:

Less NullAway errors

### CODE REVIEW

**Comment by anuraaga** [comment]
Looks like the majority are not inheriting the annotation from a base. Behavior for this doesn't seem well defined.

I'm curious how others treat this, for example does Kotlin treat all these methods as non-null? If so then adding the annotation is good, if not though maybe NullAway interpreting too strictly?

File: core/src/main/java/com/linecorp/armeria/client/Client.java [hunk 1]
```diff
@@ -71,6 +73,7 @@
      * @see ClientFactory#unwrap(Object, Class)
      * @see Unwrappable
      */
+    @Nullable
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/Client.java:76]: `@Nullable` is an inherited annotation, which means you do not have to specify it when the method you are overriding already has it, i.e. this is false positive.

File: core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java [hunk 2]
```diff
@@ -393,11 +393,13 @@ public ClientOptions options() {
 
     @Override
     public EndpointGroup endpointGroup() {
+        checkState(endpointGroup != null, "Should call init(endpoint) before invoking this method.");
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java:396]: How about using `assert`? It's perhaps a bug if this method is exposed to a user before `init()` is invoked.

File: core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotationUtil.java [hunk 3]
```diff
@@ -220,14 +220,14 @@
 
     private static <T extends Annotation> void findMetaAnnotations(
             Builder<T> builder, Annotation annotation,
-            Class<T> annotationType, Class<? extends Annotation> containerType) {
+            Class<T> annotationType, @Nullable Class<? extends Annotation> containerType) {
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotationUtil.java:223]: :+1:

**Comment by trustin** [comment]
> I'm curious how others treat this, for example does Kotlin treat all these methods as non-null? If so then adding the annotation is good, if not though maybe NullAway interpreting too strictly?

/cc Kotlin users - @gary-lo @tobias- @rfkm @okue

**Comment by ikhoon** [comment]
Kotlin also inherits `@Nullable` annotation. 

```java
// in Java
interface Super {
    @Nullable
    String foo();
}
interface NullableSub extends Super {
    @Override
    String foo();
}

interface NonNullSub extends Super {
    @NonNull
    @Override
    String foo();
}

// in Kotlin

// Nullable is inherited
val a: NullableSub = ...
val fooA: String? = a.foo()

val b: NonNullSub = ...
val fooB: String = b.foo()
```

File: core/src/main/java/com/linecorp/armeria/client/Client.java (hunk 1 above)
Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/client/Client.java]: Agreed. I reverted all `@Nullable` in override methods.

**Comment by ikhoon** [comment]
Kotlin treats non-null by default if it does have `@Nullable`.

```
// in Java
class Foo {
    @Nullable
    String bar() { ... }
    String baz() { ... }
}

// in Kotlin
val foo = Foo()
val bar: String? = foo.bar()
val baz: String = foo.baz()
```

**Comment by tobias-** [comment]
We've had issues with the Kotlin compiler in IntelliJ and the one used for Gradle exhibiting different behaviours on nonnull. Rule of thumb is that if the Gradle one accepts it, the IntelliJ one does as well. IntelliJ is more likely to accept DefaultNullable and thus causing discrepancies between the two compilations.

I will check in an hour or so.

**Comment by trustin** [comment]
Oh, it seems like we are getting assertion errors during the build. Perhaps the new `assert`s we added is the problem? :sweat_smile:

**Comment by ikhoon** [comment]
> Oh, it seems like we are getting assertion errors during the build. Perhaps the new `assert`s we added is the problem? 😅

Oops...

**Pull request merged**